lib/Pool.js

```diff
@@ -154,19 +168,22 @@ Pool.prototype.releaseConnection = function releaseConnection(connection) {
     });
   } else if (this._connectionQueue.length) {
     // get connection with next waiting callback
-    this.getConnection(this._connectionQueue.shift());
+    this.getConnection(this._connectionQueue.shift(), true);
   }
 };
 
-Pool.prototype.end = function (cb) {
-  this._closed = true;
-
+Pool.prototype.end = function (gracefulExit, cb) {
```

Comment:
So it can be expanded in the future, just make this an object instead of a plain Boolean. It'll match the signature of all the other methods that take options as well this way.